gamemodes/terrortown/entities/weapons/weapon_ttt_defibrillator.lua

```diff
@@ -277,14 +282,32 @@ if SERVER then
 			return
 		end
 
-		local spawnPoint = plyspawn.MakeSpawnPointSafe(CORPSE.GetPlayer(ent), ent:GetPos())
+		local defibAttemptResult = hook.Run("TTT2AttemptDefibPlayer", owner, ent, self)
+		if defibAttemptResult ~= nil then
+			self:Error()
+			return
+		end
+
+		local corpsePlayer = CORPSE.GetPlayer(ent)
+		if not IsValid(corpsePlayer) then
+			-- In case people want to do something about this for themselves.
+			local defibDisconnectResult = hook.Run("TTT2AttemptDefibDisconnectedPlayer", owner, ent, self)
+			if defibDisconnectResult ~= nil then
+				self:Error()
+				return
+			end
```

Comment:
I feel like this is very specific and would rather see this as a general hook in the error method to let people react to all error events.